src/browser/api/window.js

```diff
@@ -1078,21 +1078,45 @@ Window.getParentWindow = function() {};
  * Fetches window's preload script and gets its content
  */
 Window.getPreloadScript = function(identity, preloadUrl, callback) {
+    Window.setWindowPreloadState(identity, { state: 'load-started' });
```

Comment:
@whyn07m3 FYI `getPreloadScript` has been moved to api/system.js and it now just returns script from the in-memory cache. You will want to move these calls to `preload_scripts.ts`after you rebase. I can show you where therein if you like...